action/candidate_selfstake.go

```diff
@@ -0,0 +1,53 @@
+package action
+
+import (
+	"math/big"
+
+	"github.com/pkg/errors"
+
+	"github.com/iotexproject/iotex-core/pkg/version"
+)
+
+const (
+	// CandidateSelfStakeBaseIntrinsicGas represents the base intrinsic gas for CandidateSelfStake
+	CandidateSelfStakeBaseIntrinsicGas = uint64(10000)
+)
+
+// CandidateSelfStake is the action to update a candidate's bucket
+type CandidateSelfStake struct {
+	AbstractAction
+
+	// bucketID is the bucket index want to be changed to
+	bucketID uint64
```

Comment:
no `oldbucketID`?

Reply:
No need for oldBucketID, this action will attempt to set the new bucket as its own candidate self-stake bucket. As for the previous self-stake bucket, it can get from the state.